oauthproxy_test.go

```diff
@@ -843,67 +842,47 @@ func TestHttpBrowserXssFilterTrue(t *testing.T) {
-	secureMiddleware := secure.New(secure.Options{
-		AllowedHosts:            opts.HttpAllowedHosts,
-		HostsProxyHeaders:       opts.HttpHostsProxyHeaders,
-		SSLRedirect:             opts.HttpSSLRedirect,
-		SSLTemporaryRedirect:    opts.HttpSSLTemporaryRedirect,
-		SSLHost:                 opts.HttpSSLHost,
-		STSSeconds:              opts.HttpSTSSeconds,
-		STSIncludeSubdomains:    opts.HttpSTSIncludeSubdomains,
-		STSPreload:              opts.HttpSTSPreload,
-		ForceSTSHeader:          opts.HttpForceSTSHeader,
-		FrameDeny:               opts.HttpFrameDeny,
-		CustomFrameOptionsValue: opts.HttpCustomFrameOptionsValue,
-		ContentTypeNosniff:      opts.HttpContentTypeNosniff,
-		BrowserXssFilter:        opts.HttpBrowserXssFilter,
-		CustomBrowserXssValue:   opts.HttpCustomBrowserXssValue,
-		ContentSecurityPolicy:   opts.HttpContentSecurityPolicy,
-		PublicKey:               opts.HttpPublicKey,
-		ReferrerPolicy:          opts.HttpReferrerPolicy,
-	})
-	proxy := secureMiddleware.Handler(bareproxy)
+	proxy := CreateSecureProxy(opts, func(string) bool { return true })
 	rw := httptest.NewRecorder()
 	req, _ := http.NewRequest("GET", "/", nil)
 	proxy.ServeHTTP(rw, req)
 	assert.Equal(t, "1; mode=block", rw.HeaderMap.Get("X-XSS-Protection"))
 }
 
-func TestHttpBrowserXssFilterFalseBydefault(t *testing.T) {
+func TestHttpBrowserXssFilterFalseByDefault(t *testing.T) {
 	opts := NewOptions()
 	opts.ClientID = "bazquux"
 	opts.ClientSecret = "foobar"
 	opts.CookieSecret = "xyzzyplugh"
+	opts.EmailDomains = []string{"*"}
 	opts.Validate()
 
-	bareproxy := NewOAuthProxy(opts, func(string) bool { return true })
-	secureMiddleware := secure.New(secure.Options{
-		AllowedHosts:            opts.HttpAllowedHosts,
-		HostsProxyHeaders:       opts.HttpHostsProxyHeaders,
-		SSLRedirect:             opts.HttpSSLRedirect,
-		SSLTemporaryRedirect:    opts.HttpSSLTemporaryRedirect,
-		SSLHost:                 opts.HttpSSLHost,
-		STSSeconds:              opts.HttpSTSSeconds,
-		STSIncludeSubdomains:    opts.HttpSTSIncludeSubdomains,
-		STSPreload:              opts.HttpSTSPreload,
-		ForceSTSHeader:          opts.HttpForceSTSHeader,
-		FrameDeny:               opts.HttpFrameDeny,
-		CustomFrameOptionsValue: opts.HttpCustomFrameOptionsValue,
-		ContentTypeNosniff:      opts.HttpContentTypeNosniff,
-		BrowserXssFilter:        opts.HttpBrowserXssFilter,
-		CustomBrowserXssValue:   opts.HttpCustomBrowserXssValue,
-		ContentSecurityPolicy:   opts.HttpContentSecurityPolicy,
-		PublicKey:               opts.HttpPublicKey,
-		ReferrerPolicy:          opts.HttpReferrerPolicy,
-	})
-	proxy := secureMiddleware.Handler(bareproxy)
+	proxy := CreateSecureProxy(opts, func(string) bool { return true })
 	rw := httptest.NewRecorder()
 	req, _ := http.NewRequest("GET", "/", nil)
 	proxy.ServeHTTP(rw, req)
 	assert.Equal(t, false, opts.HttpBrowserXssFilter)
 	assert.Equal(t, "", rw.HeaderMap.Get("X-XSS-Protection"))
 }
+
+func TestSecureRobotsTxt(t *testing.T) {
```

Comment:
What does this test that the other tests related to `unrolled/secure` don't?

Reply:
Call all the way through the middleware into the underlying oauthproxy code, in case somehow the wrapping didn't work.